src/t_hash.c

```diff
@@ -2270,25 +2279,55 @@ static int hsetexParseArgs(client *c, int *flags,
                 return C_ERR;
 
             *expire_time_pos = i;
-        } else if (!strcasecmp(c->argv[i]->ptr, "KEEPTTL")) {
+        } else if (!strcasecmp(c->argv[i]->ptr, "PERSIST") && command_type != HASH_CMD_HSETEX) {
+            if (*flags & (HFE_EX | HFE_EXAT | HFE_PX | HFE_PXAT | HFE_PERSIST))
+                goto err_expiration;
+            *flags |= HFE_PERSIST;
+        } else if (!strcasecmp(c->argv[i]->ptr, "KEEPTTL") && command_type != HASH_CMD_HGETEX) {
             if (*flags & (HFE_EX | HFE_EXAT | HFE_PX | HFE_PXAT | HFE_KEEPTTL))
                 goto err_expiration;
             *flags |= HFE_KEEPTTL;
         } else if (!strcasecmp(c->argv[i]->ptr, "FXX")) {
+            if (command_type == HASH_CMD_HGETEX) {
+                addReplyErrorFormat(c, "unknown argument: %s", (char*) c->argv[i]->ptr);
+                return C_ERR;
+            }
```

Comment:
the check for command_type is cheaper.
```suggestion
} else if (command_type == HASH_CMD_HGETEX && !strcasecmp(c->argv[i]->ptr, "FXX")) {
```